tcms/testruns/static/testruns/js/get.js

```diff
@@ -2,4 +2,14 @@ $(document).ready(() => {
 
     $('.bootstrap-switch').bootstrapSwitch();
 
+    const testRunId = $('#test_run_pk').data('pk')
+
+    $('#status_button').on('switchChange.bootstrapSwitch', (_event, state) => {
+        if (state) {
+            jsonRPC('TestRun.update', [testRunId, { 'stop_date': null }], () => { })
+        } else {
+            const now = new Date()
+            jsonRPC('TestRun.update', [testRunId, { 'stop_date': `${now.getFullYear()}-${now.getMonth() + 1}-${now.getDate()}` }], () => { })
```

Comment:
This needs to account for hours/mines/seconds as well.

`.toISOString()` returns a string which is very close to Python's internal format and needs very little manipulation. 

And this timestamp needs to be in the timezone of the server, not locatime on the browser.

Reply:
> And this timestamp needs to be in the timezone of the server, not locatime on the browser.

this may be part of another discussion, but from my past experience it's better to handle timezones on the server side (basically save the dates along with their timestamp). UTC, is the default for the server, but what if someone configures theirs to be, say GMT+2. then the FE will be reporting dates in UTC, but the server will think they are GMT+2, no? 

that is assuming, we don't have server logic that converts the data coming from the client to the server default (which if we have, i am not aware of)

Reply:
anyhow, i implemented the change and now the FE sends the date in UTC

Reply:
Django doesn't deal well with timezone arithmetics and to make it worse both usage of timezone and the current timezone on the server can be independently configured from one another.   Anything that is a datetime input must be properly formatted and converted for the server's timezone otherwise the results will be incorrect.

When getting the current timestamp it needs to be in the server's timezone which is not always UTC. The timezone is exposed on the FE and can be made as an attribute then all calculations in JS should be in the specified timezone.